flink-core/src/main/java/org/apache/flink/api/common/io/FileInputFormat.java

```diff
@@ -819,6 +819,10 @@ public void open(FileInputSplit fileSplit) throws IOException {
 			this.stream = isot.waitForCompletion();
 			this.stream = decorateInputStream(this.stream, fileSplit);
 		}
+		catch (FileNotFoundException e) {
+			throw new FileNotFoundException("Input split " + fileSplit.getPath() +
+					" doesn't exist, skip and continue: "  + e.getMessage());
+		}
```

Comment:
As this exception doesn't have a constructor which takes a nested exception, can you use `initCause()` to patch it.

```java
throw (FileNotFoundException)(new FileNotFoundException(...)).initCause(e)
```